addons/web/models/models.py

```diff
@@ -117,6 +117,8 @@ def cleanup(vals: Dict) -> Dict:
                     vals['id']: cleanup(vals)
                     for vals in co_records.web_read(extra_fields)
                 }
+                if not many2one_data:
+                    continue
```

Comment:
You're not tackling the root cause, IMO.

Contrary to what you wrote in the description, the fact that `many2one_data` is empty has nothing to do with `extra_fields`.  In fact, when `extra_fields` is empty, you get into a special case where the output is trivially the same length as `self`.

Could it be that `co_records` are new records, and in that case, `web_read()` returns an empty list ?  IMO this shouldn't be the case, from what I understand of the code.